pkg/miscellaneous/miscellaneous.go

```diff
@@ -424,7 +424,7 @@ func IsBackupBucketEmpty(snapStoreConfig *brtypes.SnapstoreConfig, logger *logru
 	return true
 }
 
-// GetInitialClusterStateIfScaleup checks if it is the Scale-up scenario and returns the cluster state either `new` or `existing`.
+// GetInitialClusterStateIfScaleup returns `ClusterStateExisting` if etcd statefulset contains annotation `ScaledToMultiNodeAnnotationKey'.
 func GetInitialClusterStateIfScaleup(ctx context.Context, logger logrus.Entry, clientSet client.Client, podName string, podNS string) (*string, error) {
```

Comment:
Since the function returns an initial cluster state a `nil` return value does not make much sense. You should instead return `new`. Alternatively you could rename the method to `IsClusterMarkedForScaleup` and then return (bool, error)